rust/src/enhancers/rules.rs

```diff
@@ -15,6 +16,24 @@ pub struct RuleInner {
     pub actions: Vec<Action>,
 }
 
+impl fmt::Display for Rule {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        for m in &self.0.exception_matchers {
+            write!(f, "{m} ")?;
+        }
+
+        for m in &self.0.frame_matchers {
+            write!(f, "{m} ")?;
+        }
```

Comment:
As we are not maintaining source order of the caller matchers, this wouldn’t provide roundtrip idempotency. not sure if we should solve this though?

Reply:
Preserving the order here would entail keeping a list of all matchers (because sorting them into exception and frame matchers can mix up the order). I'm not sure that's worth it, to be honest.